### PR TITLE
Creating skeleton subsystems

### PR DESCRIPTION
**Pull request by SamueljWilson**

### CODE REVIEW

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -16,4 +16,29 @@ public final class Constants {
...
+  public static class DriveConstants {
+    public static final int kSmartCurrentLimit = 40;
+  }
+
+  public static class ShooterConstants {
+    //TODO: Assign real CAN ID's for the 2 flywheels
+    public static final int kLeftFlywheelID = 0;
+    public static final int kRigthFlywheelID = 0;
+
+    public static final double kDriveEncoderCPR = 2048.0;
```
Comment (keira1008) [inline src/main/java/frc/robot/Constants.java]: https://github.com/keira1008/Thor_2023/blob/465fcfb916edc36d50df0602baf779c7dc221336/src/main/java/frc/robot/Constants.java#L142 

This was recently changed.

**Review by ILIK3T0C0D3 — APPROVED** [review]
Looks great!

File: src/main/java/frc/robot/subsystems/ShooterInterp.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+    // Linear Interpolation
+    else {
+    double d0 = posTable[index-1].distance;
+    double d1 = posTable[index].distance;
+    double s0 = posTable[index-1].speed;
+    double s1 = posTable[index].speed;
+    double scaler = (distance-d0) / (d1 - d0);
+    double speedRange = s1 - s0;
+    return s0 + scaler*speedRange;
```
Comment (keira1008) [inline src/main/java/frc/robot/subsystems/ShooterInterp.java]: This isn't indented.

File: src/main/java/frc/robot/Constants.java [hunk 3]
```diff
@@ -16,4 +16,29 @@ public final class Constants {
...
+  public static class DriveConstants {
+    public static final int kSmartCurrentLimit = 40;
+  }
+
+  public static class ShooterConstants {
+    //TODO: Assign real CAN ID's for the 2 flywheels
+    public static final int kLeftFlywheelID = 0;
+    public static final int kRigthFlywheelID = 0;
+
+    public static final double kFlywheelEncoderCPR = 2048.0;
```
Comment (keira1008) [inline src/main/java/frc/robot/Constants.java]: https://github.com/keira1008/Thor_2023/blame/465fcfb916edc36d50df0602baf779c7dc221336/src/main/java/frc/robot/Constants.java#L142

This is supposed to have a different value.

**Review by keira1008 — CHANGES_REQUESTED** [review]

**Review by keira1008 — APPROVED** [review]

**Pull request merged**